Refactor compositor classes

robert-ancell opened the pull request:
Present frames directly to each view, which maintains its own compositor.
This simplifies the compositor so it doesn't need to maintain a map from views to buffers.

robert-ancell committed:
Refactor compositor classes

Present frames directly to each view, which maintains its own compositor.
This simplifies the compositor so it doesn't need to maintain a map
from views to buffers.

robert-ancell commented:
This is intended as a first step in refactoring to help land https://github.com/flutter/flutter/pull/170045

mattkae reviewed (approved):
Looks like a solid change to me :)

chinmaygarde commented:
The presubmit test failures do look real.

robert-ancell committed:
Disable failing first frame test temporarily

This now requires the widget to be realized and is proving difficult to mock.
There will be further PRs that should make this easier - reenable after they
land.

Hixie commented:
It seems this PR has introduced an intermittent bug where on my Linux machine with `xorg-server 2:21.1.12-1ubuntu1.4`, apps sometimes just stop updating their visuals.
